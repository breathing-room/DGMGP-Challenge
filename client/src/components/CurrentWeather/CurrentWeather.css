.current-container {
	width: 51.5%;
	margin: 0 auto;
	padding: 18px;
	background-color: rgba(3, 190, 190, 0.438);
	box-shadow: 2px 2px 6px -2px rgba(0,0,0,0.46);
	border-radius: 4px;
	border-bottom: 3px solid rgba(253, 116, 36, 0.521);
}

.current-main-content {
	padding: 20px;
	margin-top: -10px;
}

.current-city {
	font-weight: 100
}

.current-weather-icon {
	margin-top: -15px;
	margin-bottom: -25px;
}

.current-temp {
	font-size: 50px;
	font-weight: 350;
	margin: 0 auto;
	padding-top: 10px;
}

.current-description-container p {
	font-size: 12px;
}

.time-day {
	display: flex;
	flex-direction: row;
	margin: 0 auto;
	justify-content: center;
	margin-top: -50px;
}

.current-day {
	font-size: 22px;
	font-weight: 200;
	margin-top: 30px;
	padding: 5px;
}

.current-time {
	font-size: 22px;
	font-weight: 200;
	margin-top: 30px;
	padding: 5px;
}

.current-description {
	font-weight: 400;
	font-style: italic;
	font-size: 15px;
	margin: 0 auto;
}

.extra-details {
	display: flex;
	justify-content: center;
	margin-left: -40px;
}

.extra-details ul {
	list-style-type: none;
	font-size: 15px;
	color: #373838;
}

.extra-details ul li {
	padding: 5px;
}

.extra-details ul li span {
	font-size: 12px
}