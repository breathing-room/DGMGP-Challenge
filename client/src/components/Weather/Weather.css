.weather-container {
    padding: 20px;
    margin: 0 auto;
}

.forecast-container {
    width: 66%;
    list-style-type: none;
    display: flex;
    margin: 0 auto;
    margin-top: -20px;
    padding: 10px;
    flex-direction: row;
    justify-content: center;
}

.day {
    padding: 30px;
}