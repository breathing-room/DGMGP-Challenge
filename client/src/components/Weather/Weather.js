import React from 'react';

const Weather = () => {
    return (
        <div>Weather for a week</div>
    )
}

export default Weather;