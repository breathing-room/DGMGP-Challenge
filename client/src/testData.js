// Testing data grabbed from actual call to Open Weather API
const testData = {
	"list": 
		[
			{
				"dt": 1585677600,
				"sunrise": 1585655402,
				"sunset": 1585700293,
				"temp": {
					"day": 78.96,
					"min": 57.4,
					"max": 80.26,
					"night": 57.4,
					"eve": 68.56,
					"morn": 77.47
				},
				"feels_like": {
					"day": 72.03,
					"night": 48.29,
					"eve": 59.45,
					"morn": 76.19
				},
				"pressure": 1011,
				"humidity": 61,
				"weather": [
					{
						"id": 501,
						"main": "Rain",
						"description": "moderate rain",
						"icon": "10d"
					}
				],
				"speed": 21.21,
				"deg": 316,
				"clouds": 46,
				"rain": 3.51
			},
			{
				"dt": 1585764000,
				"sunrise": 1585741730,
				"sunset": 1585786728,
				"temp": {
					"day": 68.85,
					"min": 54.75,
					"max": 71.89,
					"night": 63.1,
					"eve": 67.75,
					"morn": 54.75
				},
				"feels_like": {
					"day": 63.81,
					"night": 61.74,
					"eve": 63.73,
					"morn": 47.53
				},
				"pressure": 1018,
				"humidity": 38,
				"weather": [
					{
						"id": 800,
						"main": "Clear",
						"description": "sky is clear",
						"icon": "01d"
					}
				],
				"speed": 5.77,
				"deg": 39,
				"clouds": 6
			},
			{
				"dt": 1585850400,
				"sunrise": 1585828058,
				"sunset": 1585873164,
				"temp": {
					"day": 75.24,
					"min": 61,
					"max": 75.42,
					"night": 65.89,
					"eve": 70.09,
					"morn": 61
				},
				"feels_like": {
					"day": 70.36,
					"night": 63.93,
					"eve": 67.01,
					"morn": 57.06
				},
				"pressure": 1018,
				"humidity": 45,
				"weather": [
					{
						"id": 803,
						"main": "Clouds",
						"description": "broken clouds",
						"icon": "04d"
					}
				],
				"speed": 10,
				"deg": 100,
				"clouds": 72
			},
			{
				"dt": 1585936800,
				"sunrise": 1585914386,
				"sunset": 1585959599,
				"temp": {
					"day": 78.55,
					"min": 63.54,
					"max": 80.28,
					"night": 69.21,
					"eve": 73.13,
					"morn": 63.54
				},
				"feels_like": {
					"day": 74.03,
					"night": 67.51,
					"eve": 69.3,
					"morn": 60.94
				},
				"pressure": 1017,
				"humidity": 36,
				"weather": [
					{
						"id": 803,
						"main": "Clouds",
						"description": "broken clouds",
						"icon": "04d"
					}
				],
				"speed": 7.85,
				"deg": 122,
				"clouds": 67
			},
			{
				"dt": 1586023200,
				"sunrise": 1586000715,
				"sunset": 1586046035,
				"temp": {
					"day": 73.24,
					"min": 65.5,
					"max": 73.69,
					"night": 66.99,
					"eve": 65.5,
					"morn": 68.34
				},
				"feels_like": {
					"day": 71.2,
					"night": 66.92,
					"eve": 62.26,
					"morn": 68.85
				},
				"pressure": 1017,
				"humidity": 63,
				"weather": [
					{
						"id": 501,
						"main": "Rain",
						"description": "moderate rain",
						"icon": "10d"
					}
				],
				"speed": 9.33,
				"deg": 73,
				"clouds": 100,
				"rain": 3.48
			}
		]
}

export default testData;