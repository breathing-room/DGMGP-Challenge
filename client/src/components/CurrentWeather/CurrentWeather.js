import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './CurrentWeather.css'

const CurrentWeather = ({ currentDay, city }) => {
      // Format current day and time
      const time = moment().format('h:hh A');
      const day = moment().format('dddd');
      // Retrieve weather icon from Open Weather Map API
      const iconUrl = `https://openweathermap.org/img/wn/${currentDay.weather[0].icon}@2x.png`
      return (
          <div className='current-container' data-testid='current-weather'> 
              <div className='current-main-content'>
                <h1 className='current-city'>{city}</h1>
                  <div className='current-weather-icon'>
                      <img src={iconUrl} alt='Weather Icon'/>
                  </div>
                  <h2 className='current-temp'>
                        {currentDay.temp.day}Â° F
                  </h2>
              </div>
              <div className='current-description-container'> 
                <span className='time-day'>
                    <h3 className='current-day'>{day}</h3>
                    <h3 className='current-time'>{time}</h3>
                </span>
                    <h3 className='current-description'>{currentDay.weather[0].description}</h3>
                    <div className='extra-details'>
                        <ul>
                            <li><b>{currentDay.feels_like.day}</b> <span>feel</span></li>
                            <li><b>{currentDay.humidity}</b> <span>humidity</span> </li>
                        </ul>
                        <ul>
                            <li><b>{currentDay.clouds}%</b> <span>cloudy</span> </li>
                            <li><b>{currentDay.speed}</b> <span>wind</span> </li>
                        </ul>
                    </div>
              </div>
          </div>
      )
}

CurrentWeather.propTypes = {
    currentDay: PropTypes.object,
    city: PropTypes.string
}

export default CurrentWeather