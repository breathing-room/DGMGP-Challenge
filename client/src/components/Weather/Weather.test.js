import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import Weather from './Weather'
import CurrentWeather from '../CurrentWeather/CurrentWeather'
import testData from '../../testData'



import { mount, shallow } from 'enzyme';

describe('Weather component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Weather />)
    })

    test('it renders correctly', () => {
            const city = 'New Orleans';
            const currentDay = testData.list[0];
            const forecast = testData.list.slice(1);
            const wrapper = shallow(
              <Weather city={city} currentDay={currentDay} forecast={forecast}/>
            );
            expect(wrapper).toMatchSnapshot();
    })
    
    test('renders CurrentWeather component', () => {
        // const { getByTestId, getAllByTestId } = render(<Weather />);
        // const weatherComponent = getByTestId('weather-component')
        // const currentWeatherComponent = within(wrapper).getAllByTestId('current-weather')
        // expect(currentWeatherComponent.length).toBe(1);

    })
})
