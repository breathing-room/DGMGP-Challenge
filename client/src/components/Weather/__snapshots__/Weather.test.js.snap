// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component it renders correctly 1`] = `ShallowWrapper {}`;
